From c6bfb5f536b0b4dbfc331018238d272262e6c8db Mon Sep 17 00:00:00 2001
From: Evgeny Grin <k2k@narod.ru>
Date: Fri, 19 May 2023 20:12:14 +0300
Subject: [PATCH] configure: fix build with ncurses

If ncurses location is not specified by configure parameters like
'--with-ncurses-includes=/some/dir' then automatic detection at compiler
default location is used. With automatic detection the header
'ncurses/term.h' is not checked therefore macro HAVE_NCURSES_TERM_H is
not defined.

Signed-off-by: Karlson2k (Evgeny Grin) <k2k@narod.ru>
---
 m4.include/mc-with-screen-ncurses.m4 | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/m4.include/mc-with-screen-ncurses.m4 b/m4.include/mc-with-screen-ncurses.m4
index 80d8478c9..a527e2308 100644
--- a/m4.include/mc-with-screen-ncurses.m4
+++ b/m4.include/mc-with-screen-ncurses.m4
@@ -134,6 +134,8 @@ AC_DEFUN([mc_WITH_NCURSES], [
             AC_MSG_ERROR([Cannot find ncurses header file])
         fi
 
+        AC_CHECK_HEADERS([ncurses/term.h])
+
         screen_type=ncurses
         screen_msg="NCurses"
         AC_DEFINE(USE_NCURSES, 1, 
-- 
2.40.1.windows.1

